### ISSUE TITLE
Unexpected duplicate id field in JSON formatted event

### ISSUE THREAD

**Issue by ernest-ag5** [cloudevents/sdk-go#772]
```
ce := cloudevents.NewEvent()
ce.SetSpecVersion("1.0")
....
ce.SetID(uuid.NewString())
ce.SetExtension("id", "bogus")
```

results in 
```
{
        	"specversion": "1.0",
                 ....
        	"id": "231f02dd-abaf-4ff3-b9e2-34ccbeb36d6b",
        	"id": "bogus"
}
```

version:  github.com/cloudevents/sdk-go/v2 v2.6.0

I think using an extension name that matches a standard field should be detected as an error (FieldErrors).
However, others formats may not have this issue.

**Comment by n3wscott** [comment]
Ah, yes. Thanks for the report. Will look at this.

**Comment by embano1** [comment]
Should be resolved, see https://github.com/cloudevents/sdk-go/pull/720 (added in `v2.7.0`)

**Comment by n3wscott** [comment]
Oh thanks @embano1 !! I thought it had been fixed but was having a hard time proving it.

Fix: update to `v2.7.0`+
